### PR TITLE
DataDog tracing 

### PR DESCRIPTION
**Pull request by abasit-folio3**

### CODE REVIEW

**Review by yvt-dynata — CHANGES_REQUESTED** [review]
We'll have to add the following to the group_vars/all.yml file:

```
  TRACING: "true"
  # used in pe-go-lib tracing
  DD_TEAM_NAME: "insightsplatform"
  # used in pe-go-lib tracing
  DD_BUSINESS_APP: "insightsplatform"
  # used in pe-go-lib tracing
  SERVICE_NAME: "oauth-proxy"
```

File: oauthproxy.go [hunk 1]
```diff
@@ -551,6 +552,9 @@ func (p *OAuthProxy) serveHTTP(rw http.ResponseWriter, req *http.Request) {
 		p.provider.Data().RedeemURL.Path)
 	req = req.Clone(ctx)
 
+	span, ctx := trace.ContextBoundTrace(ctx, "ListProjects")
```
Comment (yvt-dynata) [inline oauthproxy.go:555]: Change "ListProjects" to something more meaningful. We can use `req.URL.Path` for now.

Comment (abasit-folio3) [reply oauthproxy.go]: Done.

File: Dockerfile.debug [hunk 2]
```diff
@@ -3,6 +3,12 @@ FROM golang:1.16-buster AS builder
 # Copy sources
 WORKDIR $GOPATH/src/github.com/oauth2-proxy/oauth2-proxy
 
+
+ARG GITHUB_TOKEN
+RUN if [ -z "$GITHUB_TOKEN" ]; then echo "missing GITHUB_TOKEN env var which is needed for asset downloads"; false; fi
+# update git config to use our token for pulling assets
+RUN printf "[url \"https://%s@github.com/\"]\n\tinsteadOf = https://github.com/\n" ${GITHUB_TOKEN} >> /root/.gitconfig
+
```
Comment (yvt-dynata) [inline Dockerfile.debug:11]: We prob need to add `ARG GOPRIVATE="github.com/dynata/*,github.com/researchnow/*"` so go mod download will handle priv repository.

**Review by azfar-folio3 — APPROVED** [review]
Added GOPRIVATE ARG in Dockerfile

File: Dockerfile.debug (hunk 2 above)
Comment (azfar-folio3) [reply Dockerfile.debug:11]: Added

**Pull request closed without merging**